internal/runners/hello/hello_example.go

```diff
@@ -0,0 +1,136 @@
+// Package hello provides "hello command" logic and associated types. By
+// convention, the construction function for the primary type (also named
+// "hello") is simply named "New". For other construction functions, the name
+// of the type is used as a suffix. For instance, "hello.NewInfo()" for the
+// type "hello.Info". Similarly, "Info" would be used as a prefix or suffix for
+// the info-related types like "InfoRunParams". Each "group of types" is usually
+// expressed in its own file.
+package hello
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runbits"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+// primeable describes the app-level dependencies that a runner will need.
+type primeable interface {
+	primer.Outputer
+	primer.Auther
+	primer.Projecter
+}
+
+// RunParams defines the parameters needed to execute a given runner. These
+// values are typically collected from flags and arguments entered into the
+// cli, but there is no reason that they couldn't be set in another manner.
+type RunParams struct {
+	Name      string
+	Namespace SimpleNamespace
+	Extra     bool
+}
+
+// NewRunParams contains a scope in which default or construction-time values
+// can be set. If no default or construction-time values are necessary, direct
+// construction of RunParams is fine, and this construction func may be dropped.
+func NewRunParams() *RunParams {
+	return &RunParams{
+		Name: "Friend",
+	}
+}
+
+// Hello defines the app-level dependencies that are accessible within the Run
+// function.
+type Hello struct {
+	out output.Outputer
+	pj  *project.Project
+}
+
+// New contains the scope in which an instance of Hello is constructed from an
+// implementation of primeable.
+func New(p primeable) *Hello {
+	return &Hello{
+		out: p.Output(),
+		pj:  p.Project(),
+	}
+}
+
+// Run contains the scope in which the hello runner logic is executed.
+func (h *Hello) Run(params *RunParams) error {
+	// Reusable runner logic is contained within the runbits package.
+	// You should only use this if you intend to share logic between
+	// runners. Runners should NEVER invoke other runners.
+	if err := runbits.SayHello(h.out, params.Name); err != nil {
+		// Errors should nearly always be localized.
+		return locale.WrapError(
+			err, "hello_cannot_say", "Cannot say hello.",
+		)
+	}
```

Comment:
Can we move this after the input validation? Since this is meant to demonstrate good standards.

Reply:
The project was only necessary when the extra flag was set, but I agree with your sentiment and moved things around to do validation first (regardless of flags).